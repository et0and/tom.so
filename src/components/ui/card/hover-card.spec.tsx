import React from "react";
import { render } from "@testing-library/react";
import { HoverCard, HoverCardTrigger, HoverCardContent } from "./hover-card";

describe("<HoverCard/>", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <HoverCard>
        <HoverCardTrigger>Hover me</HoverCardTrigger>
        <HoverCardContent>
          <p>Hover card content</p>
        </HoverCardContent>
      </HoverCard>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders HoverCardTrigger correctly", () => {
    const { asFragment } = render(
      <HoverCard>
        <HoverCardTrigger>Hover me</HoverCardTrigger>
      </HoverCard>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders HoverCardContent correctly", () => {
    const { asFragment } = render(
      <HoverCard>
        <HoverCardContent>
          <p>Hover card content</p>
        </HoverCardContent>
      </HoverCard>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
