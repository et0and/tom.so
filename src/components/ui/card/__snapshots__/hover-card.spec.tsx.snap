// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HoverCard/> renders HoverCardContent correctly 1`] = `<DocumentFragment />`;

exports[`<HoverCard/> renders HoverCardTrigger correctly 1`] = `
<DocumentFragment>
  <a
    data-state="closed"
  >
    Hover me
  </a>
</DocumentFragment>
`;

exports[`<HoverCard/> renders correctly 1`] = `
<DocumentFragment>
  <a
    data-state="closed"
  >
    Hover me
  </a>
</DocumentFragment>
`;
